File: novel-library-service/src/main/java/de/roamingthings/tracing/novellibrary/domain/Novel.java
package de.roamingthings.tracing.novellibrary.domain;

import lombok.Value;

import java.time.LocalDateTime;
import java.util.UUID;

@Value
public class Novel {
    private UUID uuid;
    private LocalDateTime created;
    private LocalDateTime authored;
    private String title;
    private String content;
}

File: novel-ai-service/src/main/kotlin/de/roamingthings/tracing/novelai/domain/Novel.kt
package de.roamingthings.tracing.novelai.domain

import java.time.LocalDateTime.now
import java.time.LocalDateTime
import java.util.UUID
import java.util.UUID.fromString
import java.util.UUID.randomUUID

data class Novel(
        val uuid: NovelUuid = NovelUuid(),
        val authored: LocalDateTime = now(),
        val title: String,
        val content: String)

data class NovelUuid(val uuid: UUID = randomUUID()) {

    constructor(uuid: String) : this(fromString(uuid))

    override fun toString(): String = uuid.toString()
}
